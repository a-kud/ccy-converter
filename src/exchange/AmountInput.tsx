import React, { useState } from "react";
import styled from "styled-components";
interface AmountInputProps {
  value: string;
  inputRef: React.RefObject<HTMLInputElement>;
  onChange: (text: string) => void;
  className: string;
  placeholder: string;
}

const Input = styled("input")`
  &:focus {
    outline: none;
  }
`;
Input.displayName = "AmountInput";

export function AmountInput({
  value,
  inputRef,
  onChange,
  className,
  placeholder,
}: AmountInputProps) {
  const measureRef = React.useRef<HTMLDivElement>(null);
  const [inputWidth, setInputWidth] = useState(0);

  const handleChange = (e) => {
    onChange(e.target.value);
  };

  React.useEffect(() => {
    const { current: measureEl } = measureRef;
    if (measureEl) {
      measureEl.textContent = value;
      setInputWidth(measureEl.offsetWidth);
    }
  }, [value, inputRef, setInputWidth]);

  return (
    <Input
      dir="rtl"
      className={className}
      ref={inputRef}
      onChange={handleChange}
      value={value}
      width={inputWidth}
      placeholder={placeholder}
    />
  );
}
