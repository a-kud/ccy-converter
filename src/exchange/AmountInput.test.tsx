import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { AmountInput } from "./AmountInput";

describe("AmountInput", () => {
  const ref = {} as React.RefObject<HTMLInputElement>;
  let wrapper;
  const mockFn = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <AmountInput
        placeholder=""
        className=""
        inputRef={ref}
        onChange={mockFn}
        value="13"
      />
    );
  });

  it("renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("fires onChange callback", () => {
    const input = wrapper.find("AmountInput");
    input.props().onChange({ target: { value: "foo" } });

    expect(mockFn).toBeCalledWith("foo");
  });
});
