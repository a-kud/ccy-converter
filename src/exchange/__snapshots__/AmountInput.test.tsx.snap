// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput renders correctly 1`] = `
<AmountInput
  className=""
  dir="rtl"
  onChange={[Function]}
  placeholder=""
  value="13"
  width={0}
/>
`;
